File: Hydrogene/Nougat/OS X/BasicApplication/BasicApplication/MainWindowController.cs
namespace BasicApplication
{
	using AppKit;

	[IBObject]
	public class MainWindowController : NSWindowController
	{

    [IBOutlet]
    public NSTextField textField;

    [IBOutlet]
    public NSTextField displayMessage;

		public override id init()
		{
			this = base.initWithWindowNibName("MainWindowController");
			return this;
		}

	  [IBAction]
    public void submitButtonPressed( id sender)
    {
        this.displayMessage.stringValue = this.textField.stringValue;
    }
	}
}

File: Oxygene/Toffee/OS X/BasicApplication/BasicApplication/MainWindowController.pas
namespace BasicApplication;

interface

uses
  AppKit,
  Foundation;

type
  [IBObject]
  MainWindowController = public class(NSWindowController)
  private
  protected
  public
    [IBOutlet] textField : NSTextField;
    [IBOutlet] displayLabel: NSTextField;
    method init: id; override;
    [IBAction] method submitButtonPressed(sender : id);
  end;

implementation

method MainWindowController.init: id;
begin
  self := inherited initWithWindowNibName('MainWindowController');
  result := self;
end;


method MainWindowController.submitButtonPressed(sender: id);
begin
  self.displayLabel.stringValue := self.textField.stringValue;
end;

end.
